core/iwasm/fast-jit/fe/jit_emit_control.c

```diff
@@ -966,65 +875,11 @@ jit_compile_op_br_if(JitCompContext *cc, uint32 br_depth, uint8 **p_frame_ip)
     SET_BUILDER_POS(if_basic_block);
     SET_BB_BEGIN_BCIP(if_basic_block, *p_frame_ip - 1);
 
-    if (block_dst->label_type == LABEL_TYPE_LOOP) {
-        frame_sp_src =
-            jit_frame->sp
-            - wasm_get_cell_num(block_dst->param_types, block_dst->param_count);
-    }
-    else {
-        frame_sp_src = jit_frame->sp
-                       - wasm_get_cell_num(block_dst->result_types,
-                                           block_dst->result_count);
-    }
-
-    /* Only copy parameters or results when the src/dst addr are different */
-    copy_arities = (block_dst->frame_sp_begin != frame_sp_src) ? true : false;
-
-    if (copy_arities) {
-#if UINTPTR_MAX == UINT64_MAX
-        frame_sp_dst = jit_cc_new_reg_I64(cc);
-#else
-        frame_sp_dst = jit_cc_new_reg_I32(cc);
-#endif
-        offset = offsetof(WASMInterpFrame, lp)
-                 + (block_dst->frame_sp_begin - jit_frame->lp) * 4;
-#if UINTPTR_MAX == UINT64_MAX
-        GEN_INSN(ADD, frame_sp_dst, cc->fp_reg, NEW_CONST(I64, offset));
-#else
-        GEN_INSN(ADD, frame_sp_dst, cc->fp_reg, NEW_CONST(I32, offset));
-#endif
-    }
-
-    if (block_dst->label_type == LABEL_TYPE_LOOP) {
-        if (copy_arities) {
-            /* Dest block is Loop block, copy loop parameters */
-            copy_block_arities(cc, frame_sp_dst, block_dst->param_types,
-                               block_dst->param_count);
-        }
-        /* Jump to the begin basic block */
-        BUILD_BR(block_dst->basic_block_entry);
-        SET_BB_END_BCIP(cc->cur_basic_block, *p_frame_ip - 1);
-    }
-    else {
-        if (copy_arities) {
-            /* Dest block is Block/If/Function block, copy block results */
-            copy_block_arities(cc, frame_sp_dst, block_dst->result_types,
-                               block_dst->result_count);
-        }
-        /* Jump to the end basic block */
-        if (!(insn = GEN_INSN(JMP, 0))) {
-            jit_set_last_error(cc, "generate jmp insn failed");
-            goto fail;
-        }
-        if (!jit_block_add_incoming_insn(block_dst, insn)) {
-            jit_set_last_error(cc, "add incoming insn failed");
-            goto fail;
-        }
-        SET_BB_END_BCIP(cc->cur_basic_block, *p_frame_ip - 1);
-    }
-
-    SET_BUILDER_POS(cur_basic_block);
+    if (!handle_op_br(cc, br_depth, p_frame_ip))
```

Comment:
handle_op_br generates `JMP` instruction again, should not call it.
Had better restore the code of this function if there are no issues found.

For below wast file, the original code runs OK, but fails with this patch:

```wast
(func $1 (type $0)
    (result i32)
    i32.const 5678
    (block $l (param i32) (result i32)
      i32.const 1234
      i32.add
      i32.const 1
      br_if $l
      drop
      i32.const 666
    )
  )
```
The IRs generated by this patch:
```C
L2:
    ; PREDS(L0)
    ; BEGIN_BCIP=0x0027 END_BCIP=0x0033
    ADD             i7, 0x162e, 0x4d2
    STI32           i7, I0, 0x90
    CMP             i6, 0x1, 0x0
    BNE             i6, L3, VOID         => No ending IRs for this block
    ; SUCCS(L3 VOID)

L3:
    ; PREDS(L2)
    ; BEGIN_BCIP=0x0033 END_BCIP=0x0038
    JMP             L4
    LDI32           i8, I0, 0x90            => Unused IRs after JMP
    STI32           0x29a, I0, 0x90
    JMP             L4                           => Duplicated JMP
    ; SUCCS(L4)

L4:
```